src/layout_engine.rs

```diff
@@ -0,0 +1,206 @@
+//! The layout engine converts the layout tree to rectangles of fixed sizes on the screen,
+//! in order to know how to draw the various components.
+
+use crate::layout_abstract_syntax::*;
+use crate::layout_abstract_types::*;
+
+/// An axis along which layout is performed
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+enum Axis {
+	/// Horizontal axis, which is normally used by rows.
```

Comment:
> normally used **within** rows

(rows themselves are stacked on top of one another if there are multiple sibling rows, and sibling `<row>`s and `<column>`s are an error by the way)

Not trying to nitpick but this is a distinction that could lead to confusion (I had to think about it for a few minutes).

Reply:
I've reworded it, hopefully it makes more sense now.